File: ListDemos/app/src/main/java/org/lynxz/listdemos/adapter/RvAdapter.java
package org.lynxz.listdemos.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import org.lynxz.listdemos.R;

import java.util.ArrayList;

public class RvAdapter extends RecyclerView.Adapter<RvAdapter.MyViewHolder> {
    private final ArrayList<Integer> data;
    private final LayoutInflater mInflater;
    private static final String TAG = "RvAdapter";

    public RvAdapter(Context cxt, ArrayList<Integer> picList) {
        this.data = picList;
        mInflater = LayoutInflater.from(cxt);
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        MyViewHolder viewHolder = new MyViewHolder(mInflater.inflate(R.layout.item_rv, parent, false));
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, final int position) {
        // 设置数据
        holder.iv.setBackgroundResource(data.get(position));
        holder.tv.setText(position + "");

        // 设置事件
        holder.iv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.i(TAG, "onClick pos:" + position);
            }
        });
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    class MyViewHolder extends RecyclerView.ViewHolder {

        ImageView iv;
        TextView tv;

        public MyViewHolder(View itemView) {
            super(itemView);
            iv = (ImageView) itemView.findViewById(R.id.iv_item);
            tv = (TextView) itemView.findViewById(R.id.tv_index);
        }
    }
}

File: RecyclerViewDemo/app/src/main/java/lynxz/org/recyclerviewdemo/adapter/RvAdapter.kt
package lynxz.org.recyclerviewdemo.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import lynxz.org.recyclerviewdemo.R

/**
 * Created by Lynxz on 2016/11/16.
 * description :
 */
class RvAdapter(var cxt: Context, var data: List<String> = mutableListOf(),
                var inflater: LayoutInflater = LayoutInflater.from(cxt)) : RecyclerView.Adapter<RvAdapter.VHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VHolder {
        val view = inflater.inflate(R.layout.item, parent, false)
        return VHolder(view)
    }

    override fun getItemCount() = data.size

    override fun onBindViewHolder(holder: VHolder, position: Int) {
        holder.tv.text = data[position]
        // 简单的item单击事件,改写为 OnRecyclerItemClickListener 实现
        //        holder.itemView.setOnClickListener {
        //            Toast.makeText(cxt, "${holder.tv.text}", Toast.LENGTH_SHORT).show()
        //        }
    }

    class VHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var iv: ImageView
        var tv: TextView

        init {
            iv = itemView.findViewById(R.id.iv) as ImageView
            tv = itemView.findViewById(R.id.tv_desc) as TextView
        }
    }
}